mindsdb/api/http/namespaces/datasource.py

```diff
@@ -74,6 +76,12 @@ def get(self, name):
     @ns_conf.doc('delete_datasource')
     def delete(self, name):
         '''delete datasource'''
+
+        if not Config()["force_datasource_removing"]:
+            models = request.native_interface.get_models()
+            linked_models = [x["name"] for x in models if x['data_source_name'] == name]
+            if linked_models:
+                return "Can't delete {} datasource because there are next models linked to it: {}".format(name, linked_models), 403
         try:
             request.default_store.delete_datasource(name)
```

Comment:
It would be better to check 'force_datasource_removing' in 'delete_datasource', because datasource can be removed not only through http api

Reply:
Thanks, that is a really good note. Fixed